boa_engine/src/vm/code_block.rs

```diff
@@ -398,6 +400,355 @@ impl CodeBlock {
     }
 }
 
+impl CodeBlock {
```

Comment:
I would add this in a sub-module, in order to keep files in a reasonable size. It might also make sense to have a feature-gate for it, to avoid compiling it for normal usage.